pkg/server/backgroundsvcs/runner.go

```diff
@@ -0,0 +1,97 @@
+package backgroundsvcs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"reflect"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/grafana/dskit/services"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/modules"
+	"github.com/grafana/grafana/pkg/registry"
+)
+
+// BackgroundServiceRunner provides a runner for background services.
+type BackgroundServiceRunner struct {
+	*services.BasicService
+	registry registry.BackgroundServiceRegistry
+	log      log.Logger
+}
+
+func ProvideBackgroundServiceRunner(registry registry.BackgroundServiceRegistry) *BackgroundServiceRunner {
+	r := &BackgroundServiceRunner{registry: registry, log: log.New("background-services-runner")}
+	r.BasicService = services.NewBasicService(nil, r.run, nil).WithName(modules.BackgroundServices)
+	return r
+}
+
+func (r *BackgroundServiceRunner) run(ctx context.Context) error {
+	childRoutines, childCtx := errgroup.WithContext(ctx)
+
+	// Start background services.
+	for _, svc := range r.registry.GetServices() {
+		if registry.IsDisabled(svc) {
+			continue
+		}
+
+		service := svc
+		serviceName := reflect.TypeOf(service).String()
+		childRoutines.Go(func() error {
+			select {
+			case <-childCtx.Done():
+				return childCtx.Err()
+			default:
+			}
+			err := service.Run(childCtx)
```

Comment:
nit, we [used to log](https://github.com/grafana/grafana/blob/0d0c6401e396e6a92ae62fbd3e6059e300c8d661/pkg/server/server.go#L173) an debug message that we try to start the service: I think it would be nice to keep that